OpenSim/Common/ValueArrayDictionary.h

```diff
@@ -0,0 +1,86 @@
+/* -------------------------------------------------------------------------- *
+ *                            OpenSim:  ValueArrayDictionary.h                *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2015 Stanford University and the Authors                *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#ifndef OPENSIM_VALUEARRAYDICTIONARY_H_
+#define OPENSIM_VALUEARRAYDICTIONARY_H_
+
+#include "OpenSim/Common/ValueArray.h"
+
+#include <memory>
+
+namespace OpenSim {
+
+/** ValueArrayDictionary represents a associative array mapping from a string to
+an AbstractValueArray.                                                        */
```

Comment:
Do you want to promise anything about the performance characteristics of `ValueArrayDictionary`?